src/traces/scattergl/convert.js

```diff
@@ -460,16 +465,29 @@ proto.updateFancy = function(options) {
 };
 
 proto.updateLines = function(options, positions) {
+    var i;
     if(this.hasLines) {
-        this.lineOptions.positions = positions;
+        var linePositions = positions;
+        if(!options.connectgaps) {
+            var p = 0;
+            var x = this.xData;
+            var y = this.yData;
+            linePositions = new Float32Array(2 * x.length);
+
+            for(i=0; i<x.length; ++i) {
+                linePositions[p++] = x[i];
```

Comment:
you might have to :recycle:  this [block](https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L363-L368).